src/domain/valueObject/htmlType.go

```diff
@@ -0,0 +1,52 @@
+package valueObject
+
+import (
+	"errors"
+	"strings"
+
+	"golang.org/x/exp/slices"
+)
+
+type HtmlType string
```

Comment:
Então, HtmlType é um nome bem ruim. O ideal aqui é DataFieldType IMHO.

Reply:
Eu considerei DataFieldType, inclusive o VO foi criado com esse nome. Mudei no meio do caminho porque acredito que ficaria estranho de entender o porque estamos usando Html Types no DataFields. Falo isso considerando que, por mais que sejam tipos de DataFields, ainda assim eles são derivados do HtmlType, ou seja, se for necessário alguma pesquisa para criação de novos manifestos ou qualquer coisa relacionada a esse VO em específico, não terá confusão em relação ao que ele realmente é.

A ideia aqui é bem parecida com o HttpResponseCode nos mappings, ele é um VO relacionado aos response codes HTTP, mas faz parte dos mappings. Mesma coisa aqui.

Dito tudo isso, se ainda achar melhor que eu mude, dá um toque que eu troco, sem problemas!

Reply:
Sim, acho melhor. HttpResponseCode não tem nada a ver com HtmlTypes. Nós precisamos declarar "HttpResponseCode" na domínio pois literalmente é uma operação que a aplicação realiza, não é uma lógica de apresentação ou infra, ou seja, não é um mero detalhe, é algo core, uma função mesmo da aplicação, não tem ligação com infra, é negócio. Negócio = Entity e VOs.

Já HtmlTypes não é negócio. Ele é um auxiliar para a apresentação saber qual tipo de campo solicitar do usuário. Não é negócio pois não é uma função da aplicação e nem requerimento, é um mero detalhe.

Portanto, é uma característica de um DataField. Se é uma característica, o nome correto é DataFieldType. O fato da apresentação usar HTML é um mero detalhe que a domínio nem conhece. É uma "coincidência" usarmos os tipos de input que o Html entrega e não usamos todos. Usamos ele como base, mas não é uma relação direta. Podemos por exemplo usar o data type no CLI pra informar pro cara quais são os enums possíveis ou qual é o tipo de campo que esperamos que ele preencha. Não fazemos isso hoje, mas podemos fazer.